### PR TITLE
Timestamp local cap subs

### PR DESCRIPTION
**Pull request by RogerMartinsen**
Evaluate subscriptions based on timestamps for local subscription requests, capability post, and previous subscription request to the neighbour.

### CODE REVIEW

File: neighbour-model/src/main/java/no/vegvesen/ixn/federation/model/Capabilities.java [hunk 1]
```diff
@@ -28,8 +31,43 @@ public void setLastCapabilityExchange(LocalDateTime lastCapabilityExchange) {
...
+	}
+
+	public void setLastUpdated(LocalDateTime lastUpdated) {
+		this.lastUpdated = lastUpdated;
+	}
+
+	public void addDataType(DataType newCapability) {
+		// Add the incoming capabilities to the capabilities
+		if (dataTypes.contains(newCapability)) {
+			throw new CapabilityPostException("The posted capability already exist as capabilities. Nothing to add.");
```
Comment (kyrrek) [inline neighbour-model/src/main/java/no/vegvesen/ixn/federation/model/Capabilities.java:45]: Could we use an exception type that is more particular for the model?

File: neighbour-model/src/main/java/no/vegvesen/ixn/federation/model/Neighbour.java [hunk 2]
```diff
@@ -235,7 +236,44 @@ public void setDnsProperties(Neighbour dnsNeighbour) {
...
+	public boolean shouldCheckSubscriptionRequestsForUpdates(Optional<LocalDateTime> localSubscriptionsUpdated) {
+		if (!localSubscriptionsUpdated.isPresent()) {
+			logger.debug("Local subscriptions never established for any service provider, should not check for subscription requests");
+			return false;
+		}
+		LocalDateTime localSubscriptionsUpdatedTime = localSubscriptionsUpdated.get();
+		logger.debug("Local subscription updated {}", localSubscriptionsUpdatedTime);
+
+		LocalDateTime maxUpdateDate;
+		if (getCapabilities() != null && this.getCapabilities().getLastCapabilityExchange() != null && this.getCapabilities().getLastCapabilityExchange().isAfter(localSubscriptionsUpdatedTime)) {
```
Comment (kyrrek) [inline neighbour-model/src/main/java/no/vegvesen/ixn/federation/model/Neighbour.java]: Could this be made more explicit? Keep calling getCapabilities() and getCapabilties().getLastCapabiltyExchange over and over

**Review by kyrrek — CHANGES_REQUESTED** [review]
A lot of good work here, just a couple of small things. See comments.

File: neighbour-model/src/main/java/no/vegvesen/ixn/federation/model/Neighbour.java (hunk 2 above)
Comment (RogerMartinsen) [reply neighbour-model/src/main/java/no/vegvesen/ixn/federation/model/Neighbour.java]: see my suggestion in commit a5c3c41

**Review by kyrrek — APPROVED** [review]
Good stuff :-)

**Pull request merged**